src/dfcx_scrapi/tools/agent_checker_util.py

```diff
@@ -0,0 +1,864 @@
+            if hasattr(event_handler, "target_page") or hasattr(
+                event_handler, "target_flow"
+            ):
+                self._find_reachable_pages_rec_helper(page, event_handler,
+                                                      params)
+
+    def _process_start_page_routes_for_reachable_pages(
+        self,
+        params: Dict
+    ):
+        page = self.flow_data[params["flow_id"]]
+        for event_handler in page.event_handlers:
+            if hasattr(event_handler, "target_page") or hasattr(
+                event_handler, "target_flow"
+            ):
+                self._find_reachable_pages_rec_helper(page, event_handler,
+                                                      params)
+        for route in page.transition_routes:
+            if hasattr(route, "intent") and route.intent != "":
+                self._find_reachable_pages_rec_helper(page, route, params)
+        if params["include_groups"]:
+            for route_group in page.transition_route_groups:
+                for route in self.route_group_data[params["flow_id"]][
+                    route_group
+                ].transition_routes:
+                    if hasattr(route, "intent") and route.intent != "":
+                        self._find_reachable_pages_rec_helper(page, route,
+                                                              params)
+
+    def find_reachable_pages(
+        self,
+        flow_id: str = None,
+        flow_name: str = None,
+        from_page: str = "Start",
+        intent_route_limit: Optional[int] = None,
+        include_groups: bool = True,
+        include_start_page_routes: bool = True,
+        limit_intent_to_initial: bool = False,
+        is_initial: bool = True,
+        include_meta: bool = False,
+        verbose: bool = False,
+    ) -> List[str]:
+        """Finds all pages which are reachable by transition routes,
+        starting from a given page in a given flow. Either flow_id or
+        flow_name must be used.
+
+        Args:
+          flow_id OR flow_name: The ID or name of the flow
+          from_page: (Optional) The page to start from. If left blank, it will
+            start on the Start Page
+          intent_route_limit: (Optional) Default None
+          include_groups: (Optional) If true, intents from transition route
+            groups will be included, but only if they are actually referenced
+            on some page
+          include_start_page_routes: (Optional) Default true
+          limit_intent_to_initial: (Optional) Default False. If true, only
+            take intent routes on the initial page, rather than on any page
+            in the traversal.
+          is_initial: (Optional) Default True
+          include_meta: (Optional) Default False. If true, includes special
+            transition targets like End Session, End Flow, etc.
+          verbose: (Optional) If true, print debug information about
+            route traversal
```

Comment:
1. Could you please provide more information on `intent_route_limit` and `is_initial`?
2. It seems `include_start_page_routes` only affects the flow's Start Page (Line 519). That means if we pass some random page other than the Start page to `from_page` and pass `include_start_page_routes` it will return all reachable pages from both `from_page` and Start Page.

Reply:
1. I added some documentation about `intent_route_limit`. `is_initial` should have been internal here. I realized that I broke the functionality earlier by trying to put `is_initial` into the params dict, but I think I managed to fix it.

2. Yes, that's the intended behavior. If `include_start_page_routes` is true, then start page intent routes will be treated as if they are in scope on every page, which is the way they actually work in the agent. This isn't very useful if you're starting from a page other than the start page with no limit on the number of intent routes taken. But with `intent_route_limit = 1` for example, you can answer the question: "Starting from `from_page`, which pages can be reached in one conversation turn? (assuming all conditional routes are possible)". Choosing whether or not to allow start page routes to be in scope is pretty useful here.